src/EnergyPlus/DaylightingManager.cc

```diff
@@ -5261,128 +5133,123 @@ void GeometryTransformForDaylighting(EnergyPlusData &state)
-                    // Transform reference point coordinates into building coordinate system
-                    Xb = curRefPt.x * CosZoneRelNorth - curRefPt.y * SinZoneRelNorth + zone.OriginX;
-                    Yb = curRefPt.x * SinZoneRelNorth + curRefPt.y * CosZoneRelNorth + zone.OriginY;
-                    // Transform into World Coordinate System
-                    daylCntrl.DaylRefPtAbsCoord(1, refPtNum) = Xb * CosBldgRelNorth - Yb * SinBldgRelNorth;
-                    daylCntrl.DaylRefPtAbsCoord(2, refPtNum) = Xb * SinBldgRelNorth + Yb * CosBldgRelNorth;
-                    daylCntrl.DaylRefPtAbsCoord(3, refPtNum) = curRefPt.z + zone.OriginZ;
-                    if (doTransform) {
-                        Xo = daylCntrl.DaylRefPtAbsCoord(1, refPtNum); // world coordinates.... shifted by relative north angle...
-                        Yo = daylCntrl.DaylRefPtAbsCoord(2, refPtNum);
-                        // next derotate the building
-                        XnoRot = Xo * CosBldgRelNorth + Yo * SinBldgRelNorth;
-                        YnoRot = Yo * CosBldgRelNorth - Xo * SinBldgRelNorth;
-                        // translate
-                        Xtrans = XnoRot * std::sqrt(NewAspectRatio / OldAspectRatio);
-                        Ytrans = YnoRot * std::sqrt(OldAspectRatio / NewAspectRatio);
-                        // rerotate
-                        daylCntrl.DaylRefPtAbsCoord(1, refPtNum) = Xtrans * CosBldgRelNorth - Ytrans * SinBldgRelNorth;
-                        daylCntrl.DaylRefPtAbsCoord(2, refPtNum) = Xtrans * SinBldgRelNorth + Ytrans * CosBldgRelNorth;
-                    }
+    for (int controlNum = 1; controlNum <= state.dataDaylightingData->totDaylightingControls; ++controlNum) {
+        auto &daylCntrl = state.dataDaylightingData->daylightControl(controlNum);
+        auto &zone(state.dataHeatBal->Zone(daylCntrl.zoneIndex));
+
+        // Calc cos and sin of Zone Relative North values for later use in transforming Reference Point coordinates
+        CosZoneRelNorth = std::cos(-zone.RelNorth * DataGlobalConstants::DegToRadians);
+        SinZoneRelNorth = std::sin(-zone.RelNorth * DataGlobalConstants::DegToRadians);
+
+        rLightLevel = GetDesignLightingLevelForZone(state, daylCntrl.zoneIndex);
+        CheckLightsReplaceableMinMaxForZone(state, daylCntrl.zoneIndex);
+
+        for (refPtNum = 1; refPtNum <= daylCntrl.TotalDaylRefPoints; ++refPtNum) {
+            auto &curRefPt(state.dataDaylightingData->DaylRefPt(daylCntrl.DaylRefPtNum(refPtNum))); // get the active daylighting:referencepoint
+            curRefPt.indexToFracAndIllum = refPtNum; // back reference to the index to the ZoneDaylight structure arrays related to reference points
+            if (state.dataSurface->DaylRefWorldCoordSystem) {
+                // transform only by appendix G rotation
+                daylCntrl.DaylRefPtAbsCoord(1, refPtNum) = curRefPt.x * CosBldgRotAppGonly - curRefPt.y * SinBldgRotAppGonly;
```

Comment:
This bad girl looks like it should be an `Array1D<Vector3>` rather than `Array2D`.